### PR TITLE
Added CloudWatch client

### PR DESCRIPTION
**Pull request by GrahamCampbell**
Methods I'd like to initially support here:

- [x] `GetMetricData`
- [x] `GetMetricStatistics`
- [x] `ListMetrics`
- [x] `PutMetricData`

---

TODO:

* [x] Implementation
* [x] Unit Tests
* [ ] Integration Tests
* [x] Documentation

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Generation of `GetMetricStatistics` failed with:

```
In RestXmlParser.php line 277:
                               
  This is not implemented yet  
```

Generation of `PutMetricData` failed with:

```
In QuerySerializer.php line 167:
                                      
  Type double is not yet implemented  

```

File: docs/clients/cloud-watch-logs.md [hunk 1]
```diff
@@ -7,46 +7,24 @@ package: async-aws/cloud-watch-logs
 
 ## Usage
 
-### List log streams
+### Get metric data
```
Comment (Nyholm) [inline docs/clients/cloud-watch-logs.md:10]: I think you made a mistake modifying this file.

**Review by Nyholm — COMMENTED** [review]
Thank you. 

I think this looks good. Just update/add the tests and make sure the CI is green and I'll be happy to merge.

**Review by Nyholm — COMMENTED** [review]
Just waiting for a green CI and I'll be happy to merge.

**Comment by Nyholm** [comment]
Do you mind if I finish this PR?

**Comment by Nyholm** [comment]
Test failures are unrelated and fixed in https://github.com/async-aws/aws/pull/1079

**Pull request merged**